plugin/pkg/admission/resourcequota/config.go

```diff
@@ -63,5 +82,8 @@ func LoadConfiguration(config io.Reader) (*resourcequotaapi.Configuration, error
 	if !ok {
 		return nil, fmt.Errorf("unexpected type: %T", decodedObj)
 	}
+	// append pods as limited resources so that we can limit the number of critical pods to be created. We have
+	// a matching default quota in `kube-system` namespace which allows unlimited pods to be created in that namespace
+	resourceQuotaConfiguration.LimitedResources = append(resourceQuotaConfiguration.LimitedResources, AddCriticalPodLimitedResources()...)
```

Comment:
adding this to the user-specified config is strange to me. does that mean it is impossible to write a quota config that does not limit critical pods globally and require per-namespace quota? I'm glad it is possible to limit this globally, I just didn't expect us to force-add that to user's global quota config.

Reply:
> does that mean it is impossible to write a quota config that does not limit critical pods globally and require per-namespace quota?

I think, the limitation comes for quota object as it's limited to namespace. So, as a cluster-admin, we need to create matching quotas in every namespace that we'd like critical pods to be brought up.

Reply:
The more fundamental question is what we expect to happen whenever we add a new resource that can be controlled by global quota.

This is establishing a precedent that new resources 1) will be considered a limited resource by global quota whether or not the cluster-admin provides a global quota config file and whether or not the cluster-admin indicates they want it to be a limited resource and therefore 2) will require per-namespace quota to use

Neither of those are what I expected

Reply:
I think the scope of this `defaultQuota` should be limited to priorityClasses where it is ok to have those matching quotas per namespace which aligns with what we've in [docs](https://kubernetes.io/docs/concepts/policy/resource-quotas/#limit-priority-class-consumption-by-default). I think for other resources it's better to keep them out of `defaultQuota`.

Reply:
Also, @deads2k is suggesting to create a control loop which would ensure that we'll allow updates to only few fields in the `defaultQuota`. Having something like would also ensure that `defaultQuota` is not updated for resources other than `priorityClasses`.

Reply:
Special-casing a system-maintained quota for one type of resource in kube-system doesn't seem like it would scale well. Special-casing one type of resource to be limited-by-default and to require per-namespace quotas makes the quota policy harder to understand, and requires admins that don't care about quotaing critical pods to set up per-namespace resourcequota objects.

What is the downside to simply making the critical-pod limited resource capability available to cluster admins, without forcing them to use it?

Reply:
Given that code freeze is imminent, it might be good to jump on a call with interested parties (@ravisantoshgudimetla, @ahg-g, @derekwaynecarr, @deads2k, others?) to resolve this.

Reply:
I've to call it out that @deads2k was against special casing too. I was talking about the approach he suggested for ensuring backwards compatibility.

> What is the downside to simply making the critical-pod limited resource capability available to cluster admins, without forcing them to use it?

You mean via RBAC?

> Given that code freeze is imminent, it might be good to jump on a call with interested parties (@ravisantoshgudimetla, @ahg-g, @derekwaynecarr, @deads2k, others?) to resolve this.

Sounds good to me. Let me ping folks to slack to find time that works for everyone.

Reply:
> > What is the downside to simply making the critical-pod limited resource capability available to cluster admins, without forcing them to use it?
> 
> You mean via RBAC?

no, by specifying it explicitly in their resource quota config file in the `limitedResources` list

Reply:
Just give everyone some context. @liggit, @deads2k, @ahg-g, @damemi and I had a meeting to discuss this and we agreed up on what Jordan suggested earlier - `Have cluster-admins explicitly create a global quota object and admission config to allow which namespaces can have critical pods`